File: app/src/main/java/com/example/d/movieapp2/Adapter/Movie_Adapter.java
package com.example.d.movieapp2.Adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.d.movieapp2.R;
import com.example.d.movieapp2.data_models.Movie;
import com.example.d.movieapp2.data_models.Movie_Response;
import com.example.d.movieapp2.ui.MainActivity;
import com.squareup.picasso.Picasso;

import java.util.ArrayList;
import java.util.List;

import retrofit2.Callback;

public class Movie_Adapter extends RecyclerView.Adapter<Movie_Adapter.Movie_Holder> {

    List<Movie> movieList;

    public Movie_Adapter(List<Movie> movies, Callback<Movie_Response> callback) {
        movieList =movies;
    }


    @NonNull
    @Override
    public Movie_Holder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.movie_layout,parent,false);
        return new Movie_Holder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull Movie_Holder holder, int position) {

        holder.movieTitle.setText(movieList.get(position).getTitle());
        holder.overView.setText(movieList.get(position).getOverview());
        holder.releaseDate.setText(movieList.get(position).getReleaseDate());
        Movie movieResult = movieList.get(position);
        holder.onBind(movieResult);

//                .load("https://image.themoviedb.org/3/movie/550"+movieList
//                        .get(position).getPosterPath()).into(holder.movieImage);

    }

    @Override
    public int getItemCount() {
        return movieList.size();
    }

    public class Movie_Holder extends RecyclerView.ViewHolder {

        private ImageView movieImage;
        private TextView movieTitle;
        private TextView releaseDate;
        private TextView overView;


        public Movie_Holder(View itemView) {
            super(itemView);

            movieImage = itemView.findViewById(R.id.movie_imageview);
            movieTitle = itemView.findViewById(R.id.movie_title);
            releaseDate = itemView.findViewById(R.id.release_date);
            overView = itemView.findViewById(R.id.overview);
            movieImage = itemView.findViewById(R.id.movie_imageview);
        }

        public void onBind(Movie movieResult) {
            Picasso.with(itemView.getContext())
                    .load(movieResult.getPosterPath())
                    .into(movieImage);
        }
    }

}

File: app/src/main/java/com/example/d/movieapp2/view/MovieAdapter.kt
package com.example.d.movieapp2.view

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.navigation.Navigation
import androidx.recyclerview.widget.RecyclerView
import com.example.d.movieapp2.R
import com.example.d.movieapp2.databinding.ItemLayoutBinding
import com.example.d.movieapp2.models.MovieModel
import com.example.d.movieapp2.util.MovieClickListener

class MovieAdapter(private val movieList: ArrayList<MovieModel>) :
        RecyclerView.Adapter<MovieAdapter.MovieHolder>(), MovieClickListener {

    fun updateMovieList(newMovieList: List<MovieModel>) {
        movieList.clear()
        movieList.addAll(newMovieList)
        notifyDataSetChanged()

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MovieHolder {
        val inflater: LayoutInflater = LayoutInflater.from(parent.context)
        val view: ItemLayoutBinding =
                DataBindingUtil.inflate<ItemLayoutBinding>(inflater, R.layout.item_layout, parent
                        , false)
        return MovieHolder(view)

    }

    override fun onBindViewHolder(holder: MovieHolder, position: Int) {
        holder.view.movie = movieList[position]
        holder.view.listener = this
    }

    override fun getItemCount(): Int = movieList.size


    override fun onClick(view: View) {
        for (movie: MovieModel in movieList) {
            if (view.tag == movie.title) {
                val action =
                        MovieListFragmentDirections.actionMovieListFragmentToDetailFragment(movie)
                Navigation.findNavController(view).navigate(action)
            }

        }
    }


    class MovieHolder(var view: ItemLayoutBinding) : RecyclerView.ViewHolder(view.root)

}